src/components/modals/DefaultModal.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div class="modal-mask">
+    <div class="modal-wrapper" @click.self="$emit('close')">
+      <div class="modal-container">
+        <div class="modal-header">
+          <div class="modal-header-slot">
+            <slot name="header">default header</slot>
+          </div>
+          <span class="modal-close" @click="$emit('close')">×</span>
+        </div>
+
+        <div class="modal-body">
+          <slot name="body">default body</slot>
+        </div>
+
+        <div class="modal-footer">
+          <slot name="footer">
+            default footer
+            <button class="modal-default-button" @click="$emit('close')">OK</button>
+          </slot>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
```

Comment:
bug found, click close button top right not working. 

we can add code below to fix.

```vue
<script setup lang="ts">
const emit = defineEmits(['close']);
</script>
```
reason is close event is not relay to parent when use `DefaultModal`

Reply:
Fix this with an alternative method.
`$parent.$emit('close')`